server/src/auth/middleware.rs

```diff
@@ -0,0 +1,405 @@
+use crate::auth::error::VerifyError;
+use crate::auth::verify;
+use crate::db_connection::DBClient;
+use crate::error::CheeseError;
+use crate::models::{Token, User};
+use axum::async_trait;
+use axum::body::Body;
+use axum::extract::{FromRequest, RequestParts};
+use axum::http::header::AUTHORIZATION;
+use mongodb::bson::oid::ObjectId;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum AuthorizationError {
+    #[error("no authorization header in request")]
+    NoAuthorizationHeader,
+
+    #[error("no bearer prefix in authorization header")]
+    NoBearerPrefix,
+
+    #[error("parse token in authorization header")]
+    ParseUuid(#[from] uuid::Error),
+
+    #[error("verify")]
+    Verify(#[from] VerifyError),
+
+    #[error("admin status is required for this route")]
+    NotAdmin,
+
+    #[error("this route can only be accessed for your own user or if you are admin")]
+    NotSelf,
```

Comment:
This feels superfluous, as in, this shouldn't be a state that is accessible to a user.

Reply:
yes it can be. When you ask for a user's information but that user is not yourself. Then this error is returned.